generators/generator-base.js

```diff
@@ -2001,6 +2001,35 @@ module.exports = class extends PrivateBase {
         return constraintName;
     }
 
+    /**
+     * get a unique constraint name for tables in JHipster preferred style.
+     *
+     * @param {string} entityName - name of the entity
+     * @param {string} columnName - name of the column
+     * @param {string} prodDatabaseType - database type
+     */
+    getUXConstraintName(entityName, columnName, prodDatabaseType) {
+        const constraintName = `ux_${entityName}_${columnName}`;
```

Comment:
Is the prefix `ux` the only big difference between this method and the `getFKConstraintName` method? if so please add a param to the `getConstraintName` method to pass the prefix so that you can reuse the logic for both fk and uk

```
getConstraintName(entityName, relationshipName, prodDatabaseType, noSnakeCase, prefix = 'fk') {}
```
then you can call this from the new methods and the existing method remains backward compatible

Reply:
Unfortunately not !
`constraintName = `ux_${entityName}_${columnName}`;`
`constraintName = `fk_${entityName}_${relationshipName}_id`;`

There is a suffix in the FK constraint name... I wanted to do what you asked me, but I was blocked by this. Instead of a prefix parameter, maybe a type constraint parameter with fk as default to keep compatibility ?

Reply:
Another difference : Second parameter is relationshipName for foreign key, column name for unique constraint name... Just a manner of naming here

Reply:
Then pass the entire name as argument so that you can reuse remaining part of the method which can be moved to a new method

Reply:
I don't get it sorry :/

There are three difference in the way FK and UX constraints are named : 
- Prefix : fk and ux
- Suffix : _id and no suffix for UX
- If too long for Oracle or MySQL, fk prefix disappear... suffix follows same rules. (But OK, I can also remove the prefix for unique constraint)

So the idea is to add the type of constraint you want to name, with fk as default. Then, depending the type, give a correct name to the constraint depending the type.

The suffix part of FK name is the main big difference :/

Reply:
i'm talking about this part of the code, which is same for both the functions. You can do the snakeCase stuff for both functions as well
```
         let limit = 0;
         if (prodDatabaseType === 'oracle' && constraintName.length > 30 && !this.skipCheckLengthOfIdentifier) {
             this.warning(`The generated constraint name "${constraintName}" is too long for Oracle (which has a 30 characters limit). It will be truncated!`);
 
             limit = 28;
         } else if (prodDatabaseType === 'mysql' && constraintName.length > 64 && !this.skipCheckLengthOfIdentifier) {
             this.warning(`The generated constraint name "${constraintName}" is too long for MySQL (which has a 64 characters limit). It will be truncated!`);
 
             limit = 62;
         }
         if (limit > 0) {
             const halfLimit = Math.floor(limit / 2);
             const entityTable = noSnakeCase ? entityName.substring(0, halfLimit) : this.getTableName(entityName).substring(0, halfLimit);
             const relationTable = noSnakeCase ? relationshipName.substring(0, halfLimit - 2) : this.getTableName(relationshipName).substring(0, halfLimit - 2);
             return `${entityTable}_${relationTable}_id`;
         }
```

Reply:
Add the id in the calling function after getting the required value from the above logic. My point is I would like to avoid logic duplication. I'll make a commit on your PR with my suggestion

Reply:
```
getConstraintName(entityName, relationshipName, prodDatabaseType, noSnakeCase, prefix = FOREIGN_KEY_CONSTRAINT_PREFIX) {
        let constraintName;
        if (noSnakeCase) {
            constraintName = `${prefix}_${entityName}_${relationshipName}`;
        } else {
            constraintName = `${prefix}_${this.getTableName(entityName)}_${this.getTableName(relationshipName)}`;
        }
        if (prefix === FOREIGN_KEY_CONSTRAINT_PREFIX) {
            constraintName += '_id';
        }

        let limit = 0;

        if (prodDatabaseType === 'oracle' && constraintName.length > 30 && !this.skipCheckLengthOfIdentifier) {
            this.warning(`The generated constraint name "${constraintName}" is too long for Oracle (which has a 30 characters limit). It will be truncated!`);

            limit = 26;
        } else if (prodDatabaseType === 'mysql' && constraintName.length > 64 && !this.skipCheckLengthOfIdentifier) {
            this.warning(`The generated constraint name "${constraintName}" is too long for MySQL (which has a 64 characters limit). It will be truncated!`);

            limit = 60;
        }
        if (limit > 0) {
            const halfLimit = Math.floor(limit / 2);
            const entityTable = noSnakeCase ? entityName.substring(0, halfLimit) : this.getTableName(entityName).substring(0, halfLimit);
            const relationTable = noSnakeCase ? relationshipName.substring(0, halfLimit - 3) : this.getTableName(relationshipName).substring(0, halfLimit - 2);
            if (prefix === FOREIGN_KEY_CONSTRAINT_PREFIX) {
                return `${prefix}_${entityTable}_${relationTable}_id`;
            }
            return `${prefix}_${entityTable}_${relationTable}`;
        }
        return constraintName;
    }
```
I would do something like this. I adjusted the limit number each time.